### DISCUSSION TITLE
No refill-sets.nft files

### DISCUSSION THREAD

**Discussion by yeazzzz** [wirefalls/geo-nft#36] [General]
Ubuntu 18.04.6 LTS
nft -v
nftables v0.8.2 (Joe Btfsplk)

when i sudo geo-nft has some err happen:

Checking for settings in /etc/nftables/geo-nft/refill-sets.conf

geo-nft.sh v2.2.10:
The following 'define-ipv4' line in /etc/nftables/geo-nft/refill-sets.conf
does not point to a valid nftables set defined in your main ruleset (nftables.conf):
define-ipv4 netdev filter geo-netdev4 AD,BI

Invalid settings were found in 'refill-sets.conf'.
Automatic generation of /etc/nftables/geo-nft/refill-sets.nft will be skipped.

refill-sets.conf:
```
# Refill-sets configuration file
# Generated by geo-nft.sh v2.2.10
# Automatically flush and refill geolocation sets.
# The refill-sets.nft nftables script will be
# auto-generated using information provided here.
# Uncomment any settings below and modify as needed.
# Comment out all lines to disable this feature (default).


# Specify 'include' lines to add to refill-sets.nft.
# You can add as many include lines as required.
# ==============================================
include "/etc/nftables/geo-nft/include-all.ipv4"
include "/etc/nftables/geo-nft/include-all.ipv6"
#include "/etc/nftables/geo-nft/countrysets/*"


# Specify which sets to flush and fill with refill-sets.nft.
# Sets should be defined (without elements) in nftables.conf.
# Use 'define-ipv4' or 'define-ipv6' at the beginning of each
# line to define the IP address type. There should be 5 fields
# on each line separated by spaces. Multiple country codes
# should be comma separated without spaces between codes as
# shown below.
#
# Define-Protocol   Table Family   Table Name   Set Name   Country Codes To Fill Set
# ==================================================================================
# For 'netdev' table uncomment either or both of the following:
define-ipv4 netdev filter geo-netdev4 AD,BI
define-ipv6 netdev filter geo-netdev6 AQ,BI

# For 'inet' table (IPv4-IPv6) uncomment either or both of the following:
define-ipv4 inet filter geo-inet4 AD
define-ipv6 inet filter geo-inet6 AQ

# For 'ip' table (IPv4 only) uncomment the following:
# define-ipv4 ip filter geo-ip4 AD

# For 'ip6' table (IPv6 only) uncomment the following:
#define-ipv6 ip6 filter geo-ip6 AD

# Add any additional 'define-ipv4' or 'define-ipv6' lines here:
```

**Comment by wirefalls** [comment]
Hello,

The error message you're getting suggests that there's no nftables set defined in your `/etc/nftables.conf` file named ***geo-netdev4***. You have four set names defined in your `refill-sets.conf` file:

define-ipv4 netdev filter ***geo-netdev4*** AD,BI
define-ipv6 netdev filter ***geo-netdev6*** AQ,BI


define-ipv4 inet filter ***geo-inet4*** AD
define-ipv6 inet filter ***geo-inet6*** AQ

If you define four sets in your refill-sets.conf file then there needs to be four matching sets with the same names defined in your `/etc/nftables.conf` file. You can back up your current `/etc/nftables.conf` file and use one of the sample `nftables.conf` files from the Wiki to replace it. I would recommend starting with this one:

https://github.com/wirefalls/geo-nft/wiki/Example-IPv4-IPv6

Then follow the User Guide for set up and testing. Also, I noticed that your version of nftables is older then the minimum requirement, so I don't know if it will work or not. The README file lists nftables v0.9.0 or newer, but you can still try it out and see if it works.

I hope this helps you get things running.
